File: Sources/CountriesFlagsObjC/source/Countries/UIImage+Nauru.m
//
//  UIImage+Nauru.m
//  CountriesFlags
//

#import "Countries/UIImage+Nauru.h"

#import "Extensions/UIBezierPath+Star.h"

@implementation UIImage (Nauru)

+ (UIImage * _Nullable)flagNauruWithSize:(CGSize)size {
    UIColor* colorBlue = [UIColor colorWithRed:1.0f/255.0f
                                         green:33.0f/255.0f
                                          blue:105.0f/255.0f
                                         alpha:1.0f];
    UIColor* colorYellow = [UIColor colorWithRed:1.0f
                                           green:199.0f/255.0f
                                            blue:44.0f/255.0f
                                           alpha:1.0f];
    UIColor* colorWhite = [UIColor whiteColor];
    
    UIGraphicsBeginImageContextWithOptions(size, YES, 0.0f);
    
    [colorBlue setFill];
    UIRectFill(CGRectMake(0.0f, 0.0f, size.width, size.height));
    
    [colorYellow setFill];
    UIRectFill(CGRectMake(0.0f, size.height / 24.0f * 11.0f, size.width, size.height / 24.0f * 2.0f));
    
    [colorWhite setFill];
    [[UIBezierPath starAtPoint:CGPointMake(size.width / 48.0f * 12.0f, size.height / 24.0f * 17.0f)
                        radius:size.height / 24.0f * 4.0f
                          type:FlagStarTypeTwelvePointed] fill];
    
    UIImage* image = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    
    return image;
}

@end

File: Sources/CountriesFlags/Countries/UIImage+Nauru.swift
//
//  UIImage+Nauru.swift
//  CountriesFlags
//

#if canImport(UIKit)

import UIKit

extension UIImage {
    /**
     [wikipedia]: https://en.wikipedia.org/wiki/Flag_of_Nauru "wikipedia"
     
     Generates the flag of Nauru.
     For more information see [wikipedia].

     - Parameters:
        - size: image size.
     
     - Returns: Returns the flag of Nauru.
     */
    internal static func flagNauru(with size: CGSize) -> UIImage? {
        let colorBlue = UIColor(red: 1.0 / 255.0,
                                green: 33.0 / 255.0,
                                blue: 105.0 / 255.0,
                                alpha: 1.0)
        let colorYellow = UIColor(red: 1.0,
                                  green: 199.0 / 255.0,
                                  blue: 44.0 / 255.0,
                                  alpha: 1.0)
        let colorWhite = UIColor.white

        UIGraphicsBeginImageContextWithOptions(size, false, 0.0)

        colorBlue.setFill()
        UIRectFill(CGRect(origin: .zero, size: size))

        colorYellow.setFill()
        UIRectFill(CGRect(origin: CGPoint(x: 0.0, y: size.height / 24.0 * 11.0),
                          size: CGSize(width: size.width, height: size.height / 24.0 * 2.0)))

        colorWhite.setFill()
        let starRadius = size.height / 24.0 * 8.0 / 2.0
        UIBezierPath.star(at: CGPoint(x: size.width / 48.0 * 12.0, y: size.height / 24.0 * 17.0),
                          radius: starRadius,
                          type: .twelvePointed(innerRadius: starRadius / 2.0)).fill()

        let image = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()

        return image
    }
}

#endif
